### PR TITLE
Consistent check for masked array types

### PR DESCRIPTION
**Pull request by dkillick**
Consistency! And using part of an API for what it was meant for (in this case checking that an object is of a certain type).

To wit, anything that looks like the former will now look like the latter:

 * `if isinstance(thing, ma.core.MaskedArray)` --> `if ma.isMaskedArray(thing)`
 * `if thing is ma.masked` --> `if ma.is_masked(thing)`

### CODE REVIEW

**Comment by dkillick** [comment]
Ping @pp-mo

**Comment by dkillick** [comment]
Tests pass! Anyone want to take a look?

File: lib/iris/cube.py [hunk 1]
```diff
@@ -2221,7 +2221,7 @@ def new_cell_measure_dims(cm_):
         data = copy.deepcopy(data)
 
         # We can turn a masked array into a normal array if it's full.
-        if isinstance(data, ma.core.MaskedArray):
```
Comment (lbdreyer) [inline lib/iris/cube.py]: `ma.core.MaskedArray`!! there are quite a few different ways this has been done.

**Comment by lbdreyer** [comment]
It looks you have missed a few:
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/pandas.py#L173
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/pandas.py#L219
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/analysis/__init__.py#L1080
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/fileformats/pp.py#L1384
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/experimental/raster.py#L100

Not sure about this one:
https://github.com/dkillick/iris/blob/consistent_ma_check/lib/iris/fileformats/pp.py#L1384

File: lib/iris/cube.py (hunk 1 above)
Comment (dkillick) [inline lib/iris/cube.py]: Oh our dear old friend lack of consistency...

**Comment by dkillick** [comment]
> It looks you have missed a few

Dammit PyCharm! Your search has let me down...

**Comment by dkillick** [comment]
> Your search has let me down

But then again, perhaps we should blame my ability to regex.

**Comment by bjlittle** [comment]
@dkillick Awesome, but I think that you're still missing a couple of `core.MaskedArray` and `ma.MaskedArray` instances ...

**Comment by bjlittle** [comment]
@dkillick Let's get this merged ...

**Comment by bjlittle** [comment]
```shell
./fileformats/grib/_load_convert.py
1384:        if isinstance(data, ma.core.MaskedArray):
./fileformats/pp.py
100:    if isinstance(data, np.ma.core.MaskedArray):
./experimental/raster.py
934:    if not isinstance(a, (np.ndarray, ma.core.MaskedArray)):
./util.py
```
and
```
173:    if isinstance(data, np.ma.MaskedArray):
219:    if isinstance(data, np.ma.MaskedArray):
./pandas.py
```
These are the obvious candidates ... fix those and I'll merge.

**Comment by dkillick** [comment]
@bjlittle I've removed a few more examples...

**Pull request merged**